### PR TITLE
parity: 2.5.11 -> 2.7.2

### PR DESCRIPTION
**Pull request by bhipple**
Per comment in the upstream release notes, the `beta` track is no longer
maintained, and parity only has `stable` releases, so this deletes the beta
expression entirely.
https://github.com/paritytech/parity-ethereum/releases

This also updates to the new fetchCargoTarball implementation as part of #79975

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bhipple** [comment]
@GrahamcOfBorg build parity

**Review by akru — APPROVED** [review]
Looks good

**Comment by JohnAZoidberg** [comment]
> Looks good

Does the patch look good or have you tried to run it?

File: pkgs/applications/blockchains/parity/default.nix [hunk 1]
```diff
@@ -1,6 +1,46 @@
...
-  cargoSha256 = "16nf6y0hyffwdhxn1w4ms4zycs5lkzir8sj6c2lgsabig057hb6z";
-in
-  import ./parity.nix { inherit version sha256 cargoSha256; }
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, cmake
+, llvmPackages
+, openssl
+, pkgconfig
```
Comment (jonringer) [inline pkgs/applications/blockchains/parity/default.nix]: preference on using the canonical name, and not the alias
```suggestion
, pkg-config
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM
shows usage, but no in-depth testing

```
https://github.com/NixOS/nixpkgs/pull/80160
1 package built:
parity
```

**Review by akru — APPROVED** [review]
LGTM, thank you for your work!

**Pull request merged**